Library/ViewModels/PledgeCTAContainerViewViewModel.swift

```diff
@@ -0,0 +1,93 @@
+  var inputs: PledgeCTAContainerViewViewModelInputs { get }
+  var outputs: PledgeCTAContainerViewViewModelOutputs { get }
+}
+
+public final class PledgeCTAContainerViewViewModel: PledgeCTAContainerViewViewModelType,
+  PledgeCTAContainerViewViewModelInputs, PledgeCTAContainerViewViewModelOutputs {
+  public init() {
+    let projectAndUser = self.projectAndUserProperty.signal.skipNil()
+
+    let projectState = projectAndUser
+      .map { project, user in projectStateButton(backer: user, project: project) }
+
+    let backingEvent = projectAndUser
+      .switchMap { project, user in
+        AppEnvironment.current.apiService.fetchBacking(forProject: project, forUser: user)
+          .materialize()
+      }
+
+    let backing = backingEvent.values()
+    let project = projectAndUser.map { $0.0 }
+    let projectAndBacking = Signal.combineLatest(project, backing)
+
+    self.buttonTitleText = projectState.map { $0.buttonTitle }
+    self.buttonBackgroundColor = projectState.map { $0.buttonBackgroundColor }
+    self.stackViewIsHidden = projectState.map { $0.stackViewIsHidden }
+
+    self.rewardTitle = projectAndBacking
+      .map { (arg) -> String in
+
+        let (project, backing) = arg
+        let basicPledge = formattedAmount(for: backing)
+        let amount = Format.formattedCurrency(
+          basicPledge,
+          country: project.country,
+          omitCurrencyCode: project.stats.omitUSCurrencyCode
+        )
+        guard let rewardTitle = backing.reward?.title else { return "\(amount)" }
+        return "\(amount) • \(rewardTitle)"
+      }
+  }
+
+  fileprivate let projectAndUserProperty = MutableProperty<(Project, User)?>(nil)
+  public func configureWith(project: Project, user: User) {
+    self.projectAndUserProperty.value = (project, user)
+  }
+
+  public var inputs: PledgeCTAContainerViewViewModelInputs { return self }
+  public var outputs: PledgeCTAContainerViewViewModelOutputs { return self }
+
+  public let buttonTitleText: Signal<String, Never>
+  public let buttonBackgroundColor: Signal<UIColor, Never>
+  public let stackViewIsHidden: Signal<Bool, Never>
+  public let rewardTitle: Signal<String, Never>
+}
+
+private func projectStateButton(backer _: User, project: Project) -> PledgeStateCTAType {
+  guard let projectIsBacked = project.personalization.isBacking
+  else { return PledgeStateCTAType.viewRewards }
+
+  switch project.state {
+  case .live:
+    return projectIsBacked ? PledgeStateCTAType.manage : PledgeStateCTAType.pledge
+  case .canceled, .failed, .suspended, .successful:
+    return projectIsBacked ? PledgeStateCTAType.viewBacking : PledgeStateCTAType.viewRewards
+  default:
+    return PledgeStateCTAType.viewRewards
+  }
+}
+
+private func formattedAmount(for backing: Backing) -> String {
+  let amount = backing.amount - Double(backing.shippingAmount ?? 0)
+  let backingAmount = floor(amount) == amount
+    ? String(Int(amount))
+    : String(format: "%.2f", backing.amount)
```

Comment:
Web and Android rounds the amount (`Int(round(backing.amount))`) instead of showing decimals, so to keep consistent with the other platforms, I would do the same.

Reply:
Thanks for bringing this up! Checked in w/ @dannyalright and we're going to show the exact amount that the user pledged

Reply:
yeah. I feel like this is a case where the other platforms do it 'wrong'. Breaking platform consistency is worth it.

I'll follow up with the other platforms to see if we can get them to show the true amount.

Reply:
Just spoke to Izzy for context. Could we please make that amount reflect the **reward minimum** amount? So the number should match what the user sees on the reward tile in the next screen. 

Reward minimums never have decimals, if it helps.